### DISCUSSION TITLE
Updates And Failover Test

### DISCUSSION THREAD

**Discussion by ManarAbdulrahman** [autobase-tech/autobase#424] [Q&A] [answered]
I'm trying to test **[Type B] PostgreSQL High-Availability only** with Updates and the failover.

1- With update as it's described in read me it's a minor updates only, but specifying the minor, does not work, variable `postgrespro_minor_version` useless regarding the deployment. 

2- Failover test of the cluster with shutting down 2 nodes out of 3, so i can have only 1 node. 
but it did not work and i get errors related to etcd, is the ha applicable in this case?

**Comment by vitabaks** [comment]
> 1- With update as it's described in read me it's a minor updates only, but specifying the minor, does not work, variable postgrespro_minor_version useless regarding the deployment.

postgrespro_minor_version officially there is no such variable. Yes, only a minor update is available so far. See examples for postgrespro https://github.com/vitabaks/postgresql_cluster/issues/38

A major upgrade in the process of completing development https://github.com/vitabaks/postgresql_cluster/pull/301

> 2- Failover test of the cluster with shutting down 2 nodes out of 3, so i can have only 1 node.
but it did not work and i get errors related to etcd, is the ha applicable in this case?

Raft requires the presence of N/2 + 1, that is, more than half. 

In an etcd cluster with 3 nodes, only one node can fail. If you want to provide the possibility of failure of 2 nodes, then you need an etcd/consul cluster of 5 notes. 

Be sure to check out how it works

**Reply by ManarAbdulrahman** [reply]
so the latest of minor will be deployed for the first time, and this is applied in updates too?

**Reply by vitabaks** [reply]
maybe I didn't understand your question.

- How to update a cluster (patroni, postgres, or all system) read here https://github.com/vitabaks/postgresql_cluster#update-the-postgresql-ha-cluster
- How to upgrade postgres to a new version (14->15) - will be available soon. Or you can take part in testing https://github.com/vitabaks/postgresql_cluster/pull/301

**Reply by ManarAbdulrahman** [reply]
my point is that i want to test if updates works correctly , when i put `postgresql_version =14` it deploys the latest minor.

**Reply by vitabaks** [reply, marked as answer]
yes, this is an affordable functionality

the update will be performed to the latest available version in the repository
